### PR TITLE
[opentelemetry-operator] - Explicitly define namespace metadata on all resources

### PR DESCRIPTION
**Pull request by Allex1**
Signed-off-by: birca <Allex21ro@yahoo.com>
I think addresses https://github.com/open-telemetry/opentelemetry-helm-charts/issues/360

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]
LGTM

**Comment by TylerHelmuth** [comment]
@adomenech73 can you test if this PR solves your issue?

File: charts/opentelemetry-operator/templates/clusterrole.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ metadata:
   labels:
     app.kubernetes.io/name: opentelemetry-operator
   name: {{ template "opentelemetry-operator.name" . }}-manager-role
+  namespace: {{ .Release.Namespace }}
```
Comment (dmitryax) [inline charts/opentelemetry-operator/templates/clusterrole.yaml:9]: `ClusterRoles` are not a namespaced resource so I believe this will fail

**Review by dmitryax — APPROVED** [review]

**Review by dmitryax — CHANGES_REQUESTED** [review]
Didn't mean to approve actually

File: charts/opentelemetry-operator/templates/clusterrole.yaml (hunk 1 above)
Comment (Allex1) [reply charts/opentelemetry-operator/templates/clusterrole.yaml]: @dmitryax great catch. Thanks. Also `ClusterRoleBinding` is not a namespaced resource.

**Comment by dmitryax** [comment]
I don't think this will fix https://github.com/open-telemetry/opentelemetry-helm-charts/issues/360 . Let's discuss the issue first. helm adds namespace field automatically to all namespaced resources, so I don't think we need to do this

**Comment by Allex1** [comment]
> I don't think this will fix #360 . Let's discuss the issue first. helm adds namespace field automatically to all namespaced resources, so I don't think we need to do this

Helm does this, I agree, but depends on how the CD tool is applying the manifests. If Flux does `helm template` and then applies the  resulted manifests via `kubectl apply -f` then it will fail because of not not explicitly having the namespace defined.

**Comment by Allex1** [comment]
@dmitryax Actually both MutatingWebhookConfiguration and ValidatingWebhookConfiguration are not namespaced resources either, so closing this .

**Pull request closed without merging**